### PR TITLE
Add Missing Type Annotations

### LINKED ISSUE #18: Missing Type Annotations
There are missing type annotations for some functions in this library.

The `typing` module does not exist on CircuitPython devices so the import needs to be wrapped in try/except to catch the error for missing import. There is an example of how that is done here:

```python
try:
    from typing import List, Tuple
except ImportError:
    pass
```

Once imported the typing annotations for the argument type(s), and return type(s) can be added to the function signature. Here is an example of a function that has had this done already:

```python
def wrap_text_to_pixels(
    string: str, max_width: int, font=None, indent0: str = "", indent1: str = ""
) -> List[str]:
```
If you are new to Git or Github we have a guide about contributing to our projects here: https://learn.adafruit.com/contribute-to-circuitpython-with-git-and-github

There is also a guide that covers our CI utilities and how to run them locally to ensure they will pass in Github Actions here: https://learn.adafruit.com/creating-and-sharing-a-circuitpython-library/check-your-code In particular the pages: `Sharing docs on ReadTheDocs` and `Check your code with pre-commit` contain the tools to install and commands to run locally to run the checks.

If you are attempting to resolve this issue and need help, you can post a comment on this issue and tag both @foamyguy and @kattni or reach out to us on Discord: https://adafru.it/discord in the `#circuitpython-dev` channel.

The following locations are reported by mypy to be missing type annotations:

- [ ] adafruit_trellis.py:100
- [ ] adafruit_trellis.py:103
- [ ] adafruit_trellis.py:121
- [ ] adafruit_trellis.py:141
- [ ] adafruit_trellis.py:171
- [ ] adafruit_trellis.py:200
- [ ] adafruit_trellis.py:214
- [ ] adafruit_trellis.py:229
- [ ] adafruit_trellis.py:254
- [ ] adafruit_trellis.py:282
- [ ] adafruit_trellis.py:286
- [ ] adafruit_trellis.py:290
- [ ] adafruit_trellis.py:294

### PR DESCRIPTION
**Pull request by tcfranks**
resolves #18 
submitted for review / comment

### CODE REVIEW

File: adafruit_trellis.py [hunk 1]
```diff
@@ -197,43 +204,45 @@ def __init__(self, i2c, addresses=None):
...
-    def _write_cmd(self, byte):
+    def _write_cmd(self, byte: int) -> None:
         self._temp[0] = byte
         for device in self._i2c_devices:
             with device:
                 device.write(self._temp)
 
     @property
-    def blink_rate(self):
+    def blink_rate(self) -> int:
```
Comment (tekktrik) [inline adafruit_trellis.py:214]: This should match the setter, I think `Literal` is the way to go.

File: adafruit_trellis.py [hunk 2]
```diff
@@ -197,43 +204,45 @@ def __init__(self, i2c, addresses=None):
...
-    def brightness(self):
+    def brightness(self) -> int:
         """
         The current brightness as an integer range 0-15.
         """
         return self._brightness
 
     @brightness.setter
-    def brightness(self, brightness):
+    def brightness(
```
Comment (tekktrik) [inline adafruit_trellis.py:236]: That's a pretty big `Literal`, for the sake of readability let's just use `int`.  The real answer would be to use `Enum` but that's not currently supported by CircuitPython.

File: adafruit_trellis.py [hunk 3]
```diff
@@ -168,7 +175,7 @@ class Trellis:
 
     """
 
-    def __init__(self, i2c, addresses=None):
+    def __init__(self, i2c: I2C, addresses: Optional[int] = None) -> None:
```
Comment (tekktrik) [inline adafruit_trellis.py:178]: Looks like `addresses` should be `Optional[List[int]]`

**Review by tekktrik — CHANGES_REQUESTED** [review]
Great work!  Feedback below:

File: adafruit_trellis.py [hunk 4]
```diff
@@ -97,10 +104,10 @@
 )
 # pylint: disable=missing-docstring, protected-access
 class TrellisLEDs:
-    def __init__(self, trellis_obj):
+    def __init__(self, trellis_obj: "Trellis") -> None:
         self._parent = trellis_obj
 
-    def __getitem__(self, x):
+    def __getitem__(self, x: int) -> None:
```
[the remark this answers is not in the document]
Comment (tcfranks) [reply adafruit_trellis.py:110]: oops, I see that now

**Comment by tcfranks** [comment]
resubmitted with changes

**Review by tekktrik — APPROVED** [review]
Looks good, thanks!

**Pull request merged**